src/parser.ts

```diff
@@ -355,29 +351,31 @@ const getTokenType = (chunk: string): TokenType | undefined => {
 };
 
 const parser = (text: string, options: WordsToNumbersOptions): Region[] => {
-  const tokens = text
+  const splitText = text
+    // Split all words, spaces, and punctuation
     .split(/(\w+|\s|[[:punct:]])/i)
-    .reduce<Token[]>((acc, chunk) => {
-      const unfuzzyChunk =
-        chunk.length && options.fuzzy && !PUNCTUATION.includes(chunk)
-          ? fuzzyMatch(chunk)
-          : chunk;
-
-      const start = acc.length ? acc[acc.length - 1].end + 1 : 0;
-      const end = start + chunk.length;
-
-      if (end !== start) {
-        acc.push({
-          start,
-          end: end - 1,
-          value: unfuzzyChunk,
-          lowerCaseValue: unfuzzyChunk.toLowerCase(),
-          type: getTokenType(unfuzzyChunk),
-        });
-      }
```

Comment:
This logic was so convoluted. This was basically going through the whole flow, just to finally check if the start index and end index are the same (meaning the string is empty), where as you can just check that at the beginning or filter them out.